crates/trie/parallel/src/proof.rs

```diff
@@ -122,92 +125,103 @@ where
         // Pre-calculate storage roots for accounts which were changed.
         tracker.set_precomputed_storage_roots(storage_root_targets_len as u64);
 
-        let mut storage_proofs =
-            B256HashMap::with_capacity_and_hasher(storage_root_targets.len(), Default::default());
+        let mut storage_proofs = B256HashMap::<Vec<_>>::with_capacity_and_hasher(
+            storage_root_targets.len(),
+            Default::default(),
+        );
 
         for (hashed_address, prefix_set) in
             storage_root_targets.into_iter().sorted_unstable_by_key(|(address, _)| *address)
         {
-            let view = self.view.clone();
             let target_slots = targets.get(&hashed_address).cloned().unwrap_or_default();
-            let trie_nodes_sorted = self.nodes_sorted.clone();
-            let hashed_state_sorted = self.state_sorted.clone();
             let collect_masks = self.collect_branch_node_masks;
 
-            let (tx, rx) = std::sync::mpsc::sync_channel(1);
-
-            self.thread_pool.spawn_fifo(move || {
-                debug!(
-                    target: "trie::parallel_proof",
-                    ?hashed_address,
-                    "Starting proof calculation"
-                );
-
-                let task_start = Instant::now();
-                let result = (|| -> Result<_, ParallelStateRootError> {
-                    let provider_start = Instant::now();
-                    let provider_ro = view.provider_ro()?;
-                    trace!(
-                        target: "trie::parallel_proof",
-                        ?hashed_address,
-                        provider_time = ?provider_start.elapsed(),
-                        "Got provider"
-                    );
-
-                    let cursor_start = Instant::now();
-                    let trie_cursor_factory = InMemoryTrieCursorFactory::new(
-                        DatabaseTrieCursorFactory::new(provider_ro.tx_ref()),
-                        &trie_nodes_sorted,
-                    );
-                    let hashed_cursor_factory = HashedPostStateCursorFactory::new(
-                        DatabaseHashedCursorFactory::new(provider_ro.tx_ref()),
-                        &hashed_state_sorted,
-                    );
-                    trace!(
+            for chunk in &prefix_set
+                .into_iter()
```

Comment:
we should remove `into_iter()` or allow only using it in tests. this entirely ignores the `all` flag for self-destructed accounts